subworkflows/local/bam_variant_calling_somatic_controlfreec/main.nf

```diff
@@ -27,16 +27,39 @@ workflow BAM_VARIANT_CALLING_SOMATIC_CONTROLFREEC {
 
     FREEC_SOMATIC(controlfreec_input, fasta, fasta_fai, [], dbsnp, dbsnp_tbi, chr_files, mappability, intervals_bed, [])
 
-    ASSESS_SIGNIFICANCE(FREEC_SOMATIC.out.CNV.join(FREEC_SOMATIC.out.ratio, failOnDuplicate: true, failOnMismatch: true))
+    //Filter the files that come out of freec somatic as ASSESS_SIGNIFICANCE only takes one cnv and one ratio file
+    //Creates empty channel if file is missing
+    cnv_files = FREEC_SOMATIC.out.CNV
+    .map{ meta, cnv ->
+        def meta_clone_tumor = meta.clone()
+        def tumor_file = cnv instanceof List ? cnv.find { it.toString().endsWith("gz_CNVs") } : cnv //only find if its a list, else it returns only the filename without the path
+        def tumor = tumor_file == null ? [] : [meta_clone_tumor,tumor_file] //only fill channel if file was found, else leave it empty
```

Comment:
Yes you are right, also it doesn't make any sense to return an empty list, since that would mean there is something wrong and would be harder to debug, I modified it so that we get an error if this happens. This was also leftover code from the previous version, doesn't make sense now. Updated it.